integration_tests/structs_14b.c

```diff
@@ -0,0 +1,9 @@
+#include "structs_14b.h"
+
+int16_t sum_buffer_i16(int16_t data[], int32_t size) {
+    int16_t sum_buffer = 0;
+    for( int32_t i = 0; i < size; i++ ) {
+        sum_buffer += data[i];
+    }
+    return sum_buffer;
+}
```

Comment:
I think this is not enough. I think we need to pass the whole structure into C, then in C recreate the same structure using `char[64]` style, and then access the members and check that everything works. That would be robust.

The current test could still pass even with our pointer setup (yes, the memory would not be initialized properly, but one could get "lucky").

Reply:
Which one do you want to pass to C, `buffer_struct` (not decorated with `@ccall`) or `buffer_struct_clink` (decorated with `@ccall`)? I guess `buffer_struct_clink` should be passed to C for testing as its the one with `BindC` ABI, correct? If you want to pass both, then LLVM backend will also need to be updated to always (irrespective of the ABI) represent fixed sized array members as data only pointers

Reply:
I added test as per my understanding of the requirement. Please let me know if they look good or if you want any change in it. Thanks.